### PR TITLE
Handle invalid elliptic curve gracefully when verifying signature

### PR DESCRIPTION
**Pull request by haydentherapper**
An invalid curve caused a panic when verifying. Note that this does not affect Rekor or Fulcio. In both cases, an invalid curve will be caught when unmarshalling the PKIX public key, which checks for curve validity.

Caught by OSS-Fuzz!

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

### CODE REVIEW

File: pkg/signature/ecdsa.go [hunk 1]
```diff
@@ -181,6 +181,11 @@ func (e ECDSAVerifier) VerifySignature(signature, message io.Reader, opts ...Ver
 		return fmt.Errorf("reading signature: %w", err)
 	}
 
+	// Without this check, VerifyASN1 panics on an invalid curve.
```
Comment (bobcallaway) [inline pkg/signature/ecdsa.go:184]: is it worth a defensive check at the beginning of this function 

```
if e.publicKey == nil {
...
}
```

rare but you'd panic if that were the case

Comment (haydentherapper) [reply pkg/signature/ecdsa.go:188]: Done

**Review by bobcallaway — DISMISSED** [review]

File: pkg/signature/ecdsa.go [hunk 2]
```diff
@@ -181,6 +185,11 @@ func (e ECDSAVerifier) VerifySignature(signature, message io.Reader, opts ...Ver
...
+	if !e.publicKey.Curve.IsOnCurve(e.publicKey.X, e.publicKey.Y) {
+		return fmt.Errorf("invalid ECDSA curve for %s", e.publicKey.Params().Name)
```
Comment (asraa) [inline pkg/signature/ecdsa.go:190]: small nit: the ecdsa public key (point) is invalid (not on the given ECDSA curve)

**Review by asraa — DISMISSED** [review]
nice tests!

File: pkg/signature/ecdsa.go (hunk 2 above)
Comment (haydentherapper) [reply pkg/signature/ecdsa.go]: Thanks! Updated.

**Review by asraa — APPROVED** [review]

**Review by dlorenc — APPROVED** [review]

**Pull request merged**